### PR TITLE
Use taxonomy query instead of meta query. 

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary
 
Requires #5636 #5635 to merged. But use taxonomy query instead of meta query. 

## Testing Instructions

---

Fixes #

### CODE REVIEW

**Comment by spacedmonkey** [comment]
Once you are happy with this change, I will make sure that all tests pass.

File: includes/REST_API/Stories_Media_Controller.php [hunk 1]
```diff
@@ -171,4 +175,46 @@ protected function prepare_items_query( $prepared_args = [], $request = null ) {
...
+	 */
+	public function filter_poster_attachments( &$query ) {
+		$post_type = (array) $query->get( 'post_type' );
+
+		if ( ! in_array( 'any', $post_type, true ) && ! in_array( 'attachment', $post_type, true ) ) {
+			return;
+		}
+
+		$version = get_option( Database_Upgrader::OPTION, '0.0.0' );
+		if ( version_compare( '3.0.4', $version, '>=' ) ) {
```
Comment (swissspidy) [inline includes/REST_API/Stories_Media_Controller.php]: Once we merge #5635 (add term & migrate meta to it), we don't really need this check? Migration should have happened already? If not, when would we remove this check again?

Comment (spacedmonkey) [reply includes/REST_API/Stories_Media_Controller.php]: Sadly we don't know that. The migration runs on `admin_init`. What if the plugin is update automatically or by your hosting provider. We just never know when this database upgrade will happen. 

We could fix this issue, but hooking into [upgrader_post_install](https://github.com/WordPress/wordpress-develop/blob/e87d86b2ae1ffc48660a8fdced256ef3f94e8236/src/wp-admin/includes/class-wp-upgrader.php#L624) and force the database upgrade on plugin upgrade. But this is another ticket methinks.

Comment (swissspidy) [reply includes/REST_API/Stories_Media_Controller.php]: As soon as you open the editor, the upgrade will run. So the only scenario I see right now where this check would be needed:

1. Open editor
1. Plugin is updated in the background (e.g. by your hosting provider or in a separate tab)
1. Continue using editor -> upgrade never happens

The likelihood of that is rather small, and can be simply "fixed" by refreshing the editor.

Comment (spacedmonkey) [reply includes/REST_API/Stories_Media_Controller.php]: If you are happy then I will remove the checks and add tests to this PR.

Comment (swissspidy) [reply includes/REST_API/Stories_Media_Controller.php]: Yes please 👍

**Comment by swissspidy** [comment]
#5635 and #5636 have been merged, so this needs a rebase

**Comment by spacedmonkey** [comment]
Tests have been added.

**Pull request merged**